controllers/config_helper.go

```diff
@@ -0,0 +1,249 @@
+		config = &v1alpha1.Config{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "config",
+				Namespace: namespace,
+			},
+		}
+	} else {
+		config = getDefaultConfig(namespace)
+	}
+
+	matches := getMatches(gatekeeper)
+
+	// Append matched from gatekeeper cr spec.config.matches
+	config.Spec.Match = append(config.Spec.Match, matches...)
+
+	// Add ownerReference
+	if err := controllerutil.SetControllerReference(gatekeeper, config, scheme); err != nil {
+		return err
+	}
+
+	err := c.Create(ctx, config)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// it returns gatekeeper.spec.config.matches
+func getMatches(gatekeeper *operatorv1alpha1.Gatekeeper) (matches []v1alpha1.MatchEntry) {
+	if gatekeeper.Spec.Config != nil && gatekeeper.Spec.Config.Matches != nil {
+		// Will append gatekeeper.Spec.Config.Matches
+		matches = gatekeeper.Spec.Config.Matches
+	}
+
+	return matches
+}
+
+func hasCurrentGatekeeperUid(existingConfig *v1alpha1.Config, gatekeeper *operatorv1alpha1.Gatekeeper) bool {
+	for _, or := range existingConfig.OwnerReferences {
+		if or.UID == gatekeeper.UID {
+			return true
+		}
+	}
+
+	return false
+}
+
+// Delete previous gatekeeper ownerReference
+func deletePrevOwnerReference(existingConfig v1alpha1.Config) []metav1.OwnerReference {
+	index := -1
+
+	for i, or := range existingConfig.OwnerReferences {
+		if or.Name == "gatekeeper" {
+			index = i
+
+			break
+		}
+	}
+
+	if index == -1 {
+		return existingConfig.OwnerReferences
+	}
+
+	return append(existingConfig.OwnerReferences[:index], existingConfig.OwnerReferences[index+1:]...)
+}
+
+// Reset Config.Spec.Match and append the default exempt namespaces and
+// provided matched from gatekeeper CR
+func addDefaultExemptNamespaces(ctx context.Context,
+	c client.Client, existingConfig *v1alpha1.Config,
+	gatekeeper *operatorv1alpha1.Gatekeeper, scheme *runtime.Scheme, log logr.Logger,
+) error {
+	// Existing Config has match already and it is not created by current gatekeeper cr, don't change config
+	if !hasCurrentGatekeeperUid(existingConfig, gatekeeper) &&
+		existingConfig.Spec.Match != nil && len(existingConfig.Spec.Match) > 0 {
+		if !hasCurrentGatekeeperUid(existingConfig, gatekeeper) {
+			log.Info("The gatekeeper uid in labels does not match with the current gatekeeper uid")
+		}
+
+		if existingConfig.Spec.Match != nil && len(existingConfig.Spec.Match) > 0 {
+			log.Info("The existing config has a spec.match already. The default exempt namespace won't be attached")
+		}
+
+		return nil
+	}
+
+	// Reset existing Config Match
+	existingConfig.Spec.Match = nil
+
+	var configDefault *v1alpha1.Config
+
+	// When it is DisableDefaultMatches = false or nil
+	if gatekeeper.Spec.Config == nil || gatekeeper.Spec.Config.DisableDefaultMatches == nil ||
+		!*gatekeeper.Spec.Config.DisableDefaultMatches {
+		configDefault = getDefaultConfig("")
+		existingConfig.Spec.Match = append(existingConfig.Spec.Match, configDefault.Spec.Match[0])
```

Comment:
This will make it work if you add an additional default exempt namespace.
```suggestion
		existingConfig.Spec.Match = append(existingConfig.Spec.Match, configDefault.Spec.Match...)
```

Reply:
Thanks!! I don't know why I did `[0]`